src/Symfony/Component/Messenger/Tests/Transport/Doctrine/ConnectionTest.php

```diff
@@ -184,33 +186,38 @@ public function buildConfigurationProvider()
                     'table_name' => 'name_from_options',
                     'redeliver_timeout' => 1800,
                     'queue_name' => 'important',
+                    'auto_setup' => false,
                 ],
-                'expectedManager' => 'default',
+                'expectedConnection' => 'default',
                 'expectedTableName' => 'name_from_options',
                 'expectedRedeliverTimeout' => 1800,
                 'expectedQueue' => 'important',
+                'expectedAutoSetup' => false,
             ],
             // tests options from dsn
             [
-                'dsn' => 'doctrine://default?table_name=name_from_dsn&redeliver_timeout=1200&queue_name=normal',
+                'dsn' => 'doctrine://default?table_name=name_from_dsn&redeliver_timeout=1200&queue_name=normal&auto_setup=false',
                 'options' => [],
-                'expectedManager' => 'default',
+                'expectedConnection' => 'default',
                 'expectedTableName' => 'name_from_dsn',
                 'expectedRedeliverTimeout' => 1200,
                 'expectedQueue' => 'normal',
+                'expectedAutoSetup' => false,
             ],
             // test options from options array wins over options from dsn
             [
-                'dsn' => 'doctrine://default?table_name=name_from_dsn&redeliver_timeout=1200&queue_name=normal',
+                'dsn' => 'doctrine://default?table_name=name_from_dsn&redeliver_timeout=1200&queue_name=normal&auto_setup=true',
```

Comment:
Why not use `1` and `0` instead?

Reply:
no reason other than the docs are written with `true` and `false` everywhere. I can add another test covering that `1` and `0` are managed correctly?

Reply:
IMHO, supporting `true` and `false` is a mistake (I mean, there are strings). When using a query string, 0 and 1 is the "native" way to go.

Reply:
sure, i'm not precious either way, but note that boolean strings are also supported in the AMQP transport, i.e `amqp://localhost?persistent=true`

what do you want to do?

Reply:
do we have a consistent way to handle DSN's? something like enqueue

https://github.com/php-enqueue/enqueue-dev/blob/master/pkg/dsn/QueryBag.php#L71-L87

Reply:
I think consistency is the key here. If we are already supporting `true`/`false` elsewhere, let's support it here. Adding support for `1`/`0` would be nice as well.

We don't have a generic way of handling DSNs, but as we are now using them in many different components, it probably makes sense to create something? Could it be a new component?

Reply:
added covering test for `0/1`